arcade/arcade/core/schema.py

```diff
@@ -109,13 +110,37 @@ class ToolSecretRequirement(BaseModel):
     """The ID of the secret."""
 
 
+class ToolMetadataKey(str, Enum):
+    """Convience enum for commonly used metadata keys."""
+
+    CLIENT_ID = "client_id"
+    COORDINATOR_URL = "coordinator_url"
+
+    @staticmethod
+    def requires_auth(key: str) -> bool:
```

Comment:
naming nit: since `requires_auth` is already an important name used elsewhere in the project, maybe `needs_auth_requirement`? Just so devs don't get confused by this name.